### ISSUE TITLE
ZM8 - Kam'lanaut Fight CS issue

### ISSUE THREAD

**Issue by tagban** [DarkstarProject/darkstar#2756]
Seems to get stuck on black screen CS like the previously reported issues. This one doesn't seem fixed but you can eventually click "Exit" just not obvious. Seems like just CS is missing or ID got shifted.

**Comment by tagban** [comment]
release command doesn't work.

**Comment by dickeyf** [comment]
Just gave it a try, I'll try figure out what the exact root cause behind this.  I don't where you can click Exit though.  THis is after 5 minute wait.

**Comment by tagban** [comment]
I try my best to compile bugs from whats posted on my forums. if its not something core to DSP I don't post it here, or if its past level 75 stuff, or past CoP. I had someone post under http://bnet.cc/forums/bugs-issues-reporting.40/ exactly what was going on. I believe this one is under "Zilart"

**Comment by dickeyf** [comment]
I can reproduce it easily.  I'm trying to understand what exactly causes the client to freeze and will fix it once I understand.

**Comment by dickeyf** [comment]
I was able to fix it by letting the BCNM end only 7 seconds after the last mob is defeated.  This is closer to retail (According to many youtube videos that I watched.  The BCNM ends 7 sec after the "is defeated" message shows up).  With the 7 seconds, the client doesn't hang (When an entity_update packet with ANIMATION_DEATH is sent with the event 32001 packet, the client hangs for some reason, the 7 seconds delay make this situation impossible to happen).

Next thing I found out :
When the first param of cs 32001 is a '1', then, the client will remain black at the end of the cs (Unless skip event is chosen) albeit I can hear the character's footsteps, and the compass moves around, but the screen is black.  I believe that the '1' means that the screen must remain black because the server will send another CS after 32001 ends.  But the ending CS wasn't scripped for ZM-8.

I will submit a PR soon with these changes after I test this against 5-2, ZM8 and a BCNM :
- battlefields with no chest will end 7 seconds after all mobs are defeated to prevent the client hang.
- Will add the ending CS to ZM8, so that ZM8 becomes like retail.

Btw, I found this issue:
https://github.com/DarkstarProject/darkstar/issues/1972

Which is very similar, we should close them both once this is fixed.
